src/Http/Http.Extensions/test/RequestDelegateFactoryTests.cs

```diff
@@ -4698,6 +4698,175 @@ string HelloName(string name)
         Assert.Equal("HELLO, TESTNAMEPREFIX!", responseBody);
     }
 
+    [Fact]
+    public async Task CanInvokeFilter_OnVoidReturningHandler()
+    {
+        // Arrange
+        var invoked = false;
+        void VoidMethod()
+        {
+            invoked = true;
+        }
+        var responseBodyStream = new MemoryStream();
+        var httpContext = CreateHttpContext();
+        httpContext.Response.Body = responseBodyStream;
+
+        // Act
+        var factoryResult = RequestDelegateFactory.Create(VoidMethod, new RequestDelegateFactoryOptions()
+        {
+            RouteHandlerFilterFactories = new List<Func<RouteHandlerContext, RouteHandlerFilterDelegate, RouteHandlerFilterDelegate>>()
+            {
+                (routeHandlerContext, next) => async (context) =>
+                {
+                    return await next(context);
+                }
+            }
+        });
+        var requestDelegate = factoryResult.RequestDelegate;
+        await requestDelegate(httpContext);
+
+        Assert.Equal(200, httpContext.Response.StatusCode);
+        Assert.True(invoked);
+        var decodedResponseBody = Encoding.UTF8.GetString(responseBodyStream.ToArray());
+        Assert.Equal(string.Empty, decodedResponseBody);
+    }
+
+    [Fact]
+    public async Task CanInvokeFilter_OnTaskOfTReturningHandler()
+    {
+        // Arrange
+        var invoked = false;
+        Task<string> TaskOfTMethod()
+        {
+            invoked = true;
+            return Task.FromResult("foo");
```

Comment:
Let's add variations with `await Task.Yield();` for all the `Task/ValueTask` returning tests.